lib/inventory/metadata.go

```diff
@@ -0,0 +1,314 @@
+func (c *fetchConfig) fetchVersion() string {
+	return c.hello.Version
+}
+
+// fetchHostID returns the agent ID present in the hello message.
+func (c *fetchConfig) fetchHostID() string {
+	return c.hello.ServerID
+}
+
+// fetchOS returns the value of GOOS.
+func (c *fetchConfig) fetchOS() string {
+	return runtime.GOOS
+}
+
+// fetchServices computes the list of access protocols enabled at the agent from
+// the list of system roles present in the hello message.
+func (c *fetchConfig) fetchServices() []string {
+	var services []string
+	for _, svc := range c.hello.Services {
+		switch svc {
+		case types.RoleNode:
+			services = append(services, nodeService)
+		case types.RoleKube:
+			services = append(services, kubeService)
+		case types.RoleApp:
+			services = append(services, appService)
+		case types.RoleDatabase:
+			services = append(services, dbService)
+		case types.RoleWindowsDesktop:
+			services = append(services, windowsDesktopService)
+		}
+	}
+	return services
+}
+
+// fetchHostArchitecture computes the host architecture using the arch
+// command-line utility.
+func (c *fetchConfig) fetchHostArchitecture() string {
+	command := "arch"
+	out, err := c.exec(command)
+	if err != nil {
+		return ""
+	}
+
+	if !matchHostArchitecture.MatchString(out) {
+		return invalid(command, out)
+	}
+
+	return out
+}
+
+func (c *fetchConfig) fetchInstallMethods() []string {
+	// TODO(vitorenesduarte): fetch install methods
+	return []string{}
+}
+
+// fetchContainerRuntime returns "docker" if the file "/.dockerenv" exists.
+func (c *fetchConfig) fetchContainerRuntime() string {
+	_, err := c.read("/.dockerenv")
+	if err != nil {
+		return ""
+	}
+
+	// If the file exists, we should be running on Docker.
+	return "docker"
+}
+
+// fetchContainerOrchestrator returns kubernetes-${GIT_VERSION} if the agent is
+// running on kubernetes.
+func (c *fetchConfig) fetchContainerOrchestrator() string {
+	if c.kubeClient == nil {
+		// Return empty if not running on kubernetes.
+		return ""
+	}
+
+	version, err := c.kubeClient.Discovery().ServerVersion()
+	if err != nil {
+		log.Debugf("Failed to retrieve kubernetes server version: %s", err)
+	}
+
+	return fmt.Sprintf("kubernetes-%s", version.GitVersion)
+}
+
+// fetchCloudEnvironment returns aws, gpc or azure if the agent is running on
+// such cloud environments.
+func (c *fetchConfig) fetchCloudEnvironment() string {
+	if c.awsHttpGetSuccess() {
+		return "aws"
+	}
+	if c.gcpHttpGetSuccess() {
+		return "gcp"
+	}
+	if c.azureHttpGetSuccess() {
+		return "azure"
+	}
+	return ""
+}
+
+// awsHttpGetSuccess hits the AWS metadata endpoint in order to detect whether
+// the agent is running on AWS.
+// https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-retrieval.html
+func (c *fetchConfig) awsHttpGetSuccess() bool {
+	url := "http://169.254.169.254/latest/meta-data/"
+	req, err := http.NewRequest(http.MethodGet, url, nil)
```

Comment:
Here and below.

```suggestion
	req, err := http.NewRequestWithContext(c.ctx, http.MethodGet, url, nil)
```

Reply:
Thanks! Done in 883117cd82b37a276dd6ec1e957fb1cee9743ec9.